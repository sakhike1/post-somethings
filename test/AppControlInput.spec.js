import { mount } from "@vue/test-utils";
import AppControlInput from "@/components/UI/AppControlInput.vue";

describe("AppControlInput", () => {
  it('renders an input control when controlType is "input"', () => {
    const wrapper = mount(AppControlInput, {
      propsData: {
        controlType: "input",
        value: "Hello",
      },
    });

    const inputControl = wrapper.find("input");
    expect(inputControl.exists()).toBe(true);
    expect(inputControl.element.value).toBe("Hello");
  });

  it('renders a textarea control when controlType is "textarea"', () => {
    const wrapper = mount(AppControlInput, {
      propsData: {
        controlType: "textarea",
        value: "Hello",
      },
    });

    const textareaControl = wrapper.find("textarea");
    expect(textareaControl.exists()).toBe(true);
    expect(textareaControl.element.value).toBe("Hello");
  });

  it("emits input event when the input control value changes", async () => {
    const wrapper = mount(AppControlInput, {
      propsData: {
        controlType: "input",
        value: "Hello",
      },
    });

    const inputControl = wrapper.find("input");
    await inputControl.setValue("Updated value");

    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input[0][0]).toBe("Updated value");
    expect().toMatchSnapshot();
  });
});
