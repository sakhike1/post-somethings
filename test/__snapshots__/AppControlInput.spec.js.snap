// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppControlInput emits input event when the input control value changes 1`] = `undefined`;
